src/audio_resampler.h

```diff
@@ -67,16 +67,51 @@ class AudioResampler : public AudioDecoder {
 
 	/**
 	 * Wraps the opening function of the contained decoder
-	 * 
+	 *
 	 * @param[in] file Filepointer to a file readable by the wrapped decoder
-	 * 
+	 *
 	 * @return Whether the operation was successful or not
 	 */
 	bool Open(Filesystem_Stream::InputStream stream) override;
 
+	/**
+	 * Forwards a pause to the wrapped decoder.
+	 */
+	void Pause() override;
+
+	/**
+	 * Forwards a resume to the wrapped decoder.
+	 */
+	void Resume() override;
+
+	/**
+	 * Obtains the volume of the wrapped decoder.
+	 *
+	 * @return current volume (from 0 - 100)
+	 */
+	int GetVolume() const override;
+
+	/**
+	 * Sets the current volume of the wrapped decoder.
+	 *
+	 * @param volume (from 0-100)
+	 */
+	void SetVolume(int volume) override;
+
+	/**
+	 * Prepares a volume fade in/out effect.
+	 * To do a fade out begin must be larger then end.
+	 * Call Update to do the fade.
+	 *
+	 * @param begin Begin volume (from 0-100)
+	 * @param end End volume (from 0-100)
+	 * @param duration Fade duration in ms
+	 */
+	void SetFade(int begin, int end, std::chrono::milliseconds duration) override;
+
 	/**
 	 * Wraps the seek function of the contained decoder
-	 * @note If the seek function of the wrapped decoder is 
+	 * @note If the seek function of the wrapped decoder is
 	 *	somewhat corelated to time the offset is not influenced by the resampling ratio
```

Comment:
Nit: This existed before, but tab character nesting note could use spaces after the asterisk for better aligning.